### PR TITLE
Suppport qwen model and solve some problems

### PR DESCRIPTION
**Pull request by Arcmoon-Hu**
1、Add Qwen Model support based on vllm's qwen model file
2、Solve some problems when loading Qwen model

### CODE REVIEW

File: python/sglang/srt/managers/router/model_runner.py [hunk 1]
```diff
@@ -258,6 +259,8 @@ def load_model(self):
             if arch == "MixtralForCausalLM":
                 model_class = MixtralForCausalLM
                 break
+            if arch == "QWenLMHeadModel":
+                model_class = QWenLMHeadModel
```
Comment (merrymercy) [inline python/sglang/srt/managers/router/model_runner.py:263]:
```suggestion
                model_class = QWenLMHeadModel
                break
```

File: python/sglang/srt/model_config.py [hunk 2]
```diff
@@ -20,8 +20,11 @@ def __init__(
         # Unify the config keys for hf_config
         self.context_len = get_context_length(self.hf_config)
         self.head_dim = self.hf_config.hidden_size // self.hf_config.num_attention_heads
-        self.num_key_value_heads = self.hf_config.num_key_value_heads
         self.num_attention_heads = self.hf_config.num_attention_heads
+        try:
+            self.num_key_value_heads = self.hf_config.num_key_value_heads
+        except Exception as e:
```
Comment (merrymercy) [inline python/sglang/srt/model_config.py]: do not use a general exception. Can you try something like `KeyNotFoundError`

**Review by merrymercy — CHANGES_REQUESTED** [review]

File: python/sglang/srt/model_config.py (hunk 2 above)
Comment (Arcmoon-Hu) [reply python/sglang/srt/model_config.py]: I have change it by using `getattr` method

File: python/sglang/srt/model_config.py [hunk 3]
```diff
@@ -21,9 +21,8 @@ def __init__(
         self.context_len = get_context_length(self.hf_config)
         self.head_dim = self.hf_config.hidden_size // self.hf_config.num_attention_heads
         self.num_attention_heads = self.hf_config.num_attention_heads
-        try:
-            self.num_key_value_heads = self.hf_config.num_key_value_heads
-        except Exception as e:
+        num_key_value_heads = getattr(self.hf_config, "num_key_value_heads", None)
```
Comment (cyLi-Tiger) [inline python/sglang/srt/model_config.py]: should be `self.num_key_value_heads`?

File: python/sglang/srt/model_config.py [hunk 4]
```diff
@@ -20,8 +20,10 @@ def __init__(
         # Unify the config keys for hf_config
         self.context_len = get_context_length(self.hf_config)
         self.head_dim = self.hf_config.hidden_size // self.hf_config.num_attention_heads
-        self.num_key_value_heads = self.hf_config.num_key_value_heads
         self.num_attention_heads = self.hf_config.num_attention_heads
+        num_key_value_heads = getattr(self.hf_config, "num_key_value_heads", None)
+        if num_key_value_heads is None:
```
Comment (merrymercy) [inline python/sglang/srt/model_config.py]:
```suggestion
        self.num_key_value_heads = getattr(self.hf_config, "num_key_value_heads", None)
        if self.num_key_value_heads is None:
```

**Comment by merrymercy** [comment]
@Arcmoon-Hu Thanks for the contribution! It will be great if you can also follow Llava and add the support for QwenVL!

File: python/sglang/srt/model_config.py (hunk 3 above)
Comment (Arcmoon-Hu) [reply python/sglang/srt/model_config.py]: Yes, it is my negligence. Sorry

**Comment by Arcmoon-Hu** [comment]
Thanks for your advice. I will continue to follow this project and hope it gets better and better

**Pull request merged**